pylint/checkers/similar.py

```diff
@@ -195,6 +201,16 @@ def stripped_lines(lines, ignore_comments, ignore_docstrings, ignore_imports):
             )
         }
         current_line_is_import = False
+    if ignore_signatures:
+        functions = filter(
+            lambda node: isinstance(
+                node, (astroid.FunctionDef, astroid.AsyncFunctionDef)
+            ),
+            tree.body,
+        )
```

Comment:
```suggestion
        functions = [
            n for n in tree.body if  isinstance(n, (astroid.FunctionDef, astroid.AsyncFunctionDef))
        ]
```
We have a check for that, (prefer list comprehension over filter/lambda/map) but I think it's deactivated right now. I'll have a look later on.